core/execution/market.go

```diff
@@ -774,6 +774,47 @@ func (m *Market) updateMarketValueProxy() {
 	m.stateChanged = true
 }
 
+func (m *Market) closeCancelledMarket(ctx context.Context, t time.Time) error {
+	m.tradableInstrument.Instrument.Unsubscribe(ctx)
+
+	asset, _ := m.mkt.GetAsset()
+	parties := make([]string, 0, len(m.parties))
+	for k := range m.parties {
+		parties = append(parties, k)
+	}
+
+	sort.Strings(parties)
+	clearMarketTransfers, err := m.collateral.ClearMarket(ctx, m.GetID(), asset, parties)
+	if err != nil {
+		m.log.Error("Clear market error",
+			logging.MarketID(m.GetID()),
+			logging.Error(err))
+		return err
+	}
+
+	// unregister state-variables
+	m.stateVarEngine.UnregisterStateVariable(asset, m.mkt.ID)
+
+	m.broker.Send(events.NewTransferResponse(ctx, clearMarketTransfers))
+	m.mkt.TradingMode = types.MarketTradingModeNoTrading
+	m.broker.Send(events.NewMarketUpdatedEvent(ctx, *m.mkt))
+	m.closed = true
+	m.mkt.State = types.MarketStateCancelled
```

Comment:
should we not also have here:

`m.broker.Send(events.NewMarketUpdatedEvent(ctx, *m.mkt))`